merlin/table/tensor_column.py

```diff
@@ -0,0 +1,155 @@
+#
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Type
+
+import merlin.dtypes as md
+from merlin.core.protocols import SeriesLike
+
+
+class Device(Enum):
+    CPU = 0
+    GPU = 1
+
+
+# This should always contains arrays or tensors, not series
+class TensorColumn(ABC, SeriesLike):
```

Comment:
`TensorColumn` should be `SeriesLike`, but I don't think we can explicitly declare it here and still appease the linter. Maybe we could add tests that check that `isinstance(TensorColumn(), SeriesLike)` and `isinstance(TensorTable(), DataFrameLike)`?